Project2015To2017/PatternProcessor.cs

```diff
@@ -0,0 +1,4 @@
+namespace Project2015To2017
+{
+	public delegate bool PatternProcessor(ProjectConverter converter, string pattern, ProcessSingleItemCallback callback, Facility self);
```

Comment:
Out of curiosity, any reason why you're not using a Func?

Reply:
They are mostly interchangeable, but delegate provides more explicit function signature _declaration_.